pkg/filters/proxy/metrics.go

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package proxy
+
+import (
+	"github.com/megaease/easegress/pkg/protocols/httpprot/httpstat"
+	"github.com/megaease/easegress/pkg/util/prometheushelper"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type (
+	// ProxyMetrics is the Prometheus Metrics of ProxyMetrics Object.
+	metrics struct {
+		TotalConnections      *prometheus.CounterVec
+		TotalErrorConnections *prometheus.CounterVec
+		RequestBodySize       prometheus.ObserverVec
+		ResponseBodySize      prometheus.ObserverVec
+	}
+)
+
+// newMetrics create the ProxyMetrics.
+func (p *Proxy) newMetrics(name string) *metrics {
```

Comment:
propose to put all the content of this file into pool.go, and change the receiver of this function to `ServerPool`.